File: pdu/python/std_msgs/pdu_cdr_conv_UInt8.py
from .pdu_pytype_UInt8 import UInt8
from ..pdu_cdr_runtime import CdrReader, CdrWriter




def py_to_cdr_body_UInt8(writer: CdrWriter, src: UInt8):
    writer.write_uint8(src.data)


def cdr_body_to_py_UInt8(reader: CdrReader, dst: UInt8):
    dst.data = reader.read_uint8()
    return dst


def py_to_cdr_UInt8(src: UInt8) -> bytes:
    writer = CdrWriter()
    writer.write_encapsulation()
    py_to_cdr_body_UInt8(writer, src)
    return writer.bytes()


def cdr_to_py_UInt8(cdr_payload) -> UInt8:
    reader = CdrReader(cdr_payload)
    reader.read_encapsulation()
    dst = UInt8()
    return cdr_body_to_py_UInt8(reader, dst)

File: pdu/javascript/std_msgs/pdu_cdr_conv_UInt8.js
import { UInt8 } from './pdu_jstype_UInt8.js';
import { PduCdrWriter, PduCdrReader } from '../pdu_cdr_runtime.js';


export class PduUInt8Converter {
    /**
     * @param {PduCdrWriter} writer
     * @param { UInt8 } src
     */
    static to_cdr_body(writer, src) {
        writer.write_uint8(src.data);
    }

    /**
     * @param {PduCdrReader} reader
     * @param { UInt8 } dst
     * @returns { UInt8 }
     */
    static cdr_body_to_js(reader, dst) {
        dst.data = reader.read_uint8();
        return dst;
    }

    /**
     * @param { UInt8 } src
     * @returns {ArrayBuffer}
     */
    static to_cdr(src) {
        const writer = new PduCdrWriter();
        writer.write_encapsulation();
        this.to_cdr_body(writer, src);
        return writer.get_buf();
    }

    /**
     * @param {ArrayBuffer|ArrayBufferView} cdrPayload
     * @returns { UInt8 }
     */
    static from_cdr(cdrPayload) {
        const reader = new PduCdrReader(cdrPayload);
        reader.read_encapsulation();
        return this.cdr_body_to_js(reader, new UInt8());
    }
}
